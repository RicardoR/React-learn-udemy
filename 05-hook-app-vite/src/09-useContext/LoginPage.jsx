import {useContext} from "react";
import {UserContext} from "./context/UserContext";

export const LoginPage = () => {

    const {user, setUser} = useContext(UserContext);

    const loginUser = () => {
        const user = {
            id: 123,
            name: 'Ricardo',
            mail: 'ricardo@mail'
        }
        setUser(user);
    }

    return (
        <>
            <h1> LoginPage</h1>
            <hr/>
            <pre>
                {JSON.stringify(user, null, 3)}
            </pre>
            <button
                onClick={loginUser}
                className='btn btn-primary'>
                Set user
            </button>
        </>
    );
};

