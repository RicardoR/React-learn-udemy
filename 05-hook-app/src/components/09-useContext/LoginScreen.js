import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext);

  const handleLogin = () => {
    setUser({
      id: 123,
      name: 'John Doe',
    });
  };

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 3)}</pre>

      <button className="btn btn-primary" onClick={() => handleLogin()}>
        Login
      </button>
    </div>
  );
};
