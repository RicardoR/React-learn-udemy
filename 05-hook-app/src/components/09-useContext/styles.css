.navbar {
  padding: 0.5rem;
  margin-bottom: 24px;
}
