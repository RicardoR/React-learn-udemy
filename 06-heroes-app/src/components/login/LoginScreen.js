import { useNavigate } from 'react-router-dom';

export const LoginScreen = () => {
  const navigate = useNavigate();
  const handleLogin = () => {
    // replace true evita que puedas volver atr√°s a esta pantalla
    navigate('/', { replace: true });
  };

  return (
    <div className="container mt-5">
      <h1>LoginScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={handleLogin}>
        Login
      </button>
    </div>
  );
};
