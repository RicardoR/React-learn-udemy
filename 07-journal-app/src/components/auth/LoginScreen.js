import React, { useState } from 'react';
import validator from 'validator';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import {
  startLoginWithEmailPassword,
  startGoogleLogin,
} from '../../actions/auth';
import { useForm } from '../../hooks/useForm';

export const LoginScreen = () => {
  const dispatch = useDispatch();
  const [error, setError] = useState(undefined);
  const { msgError, loading } = useSelector((state) => state.ui);

  const [formValues, handleInputChange] = useForm({
    email: 'email@gmail.com',
    password: '123456',
  });

  const { email, password } = formValues;

  const handleLogin = (event) => {
    event.preventDefault();
    if (isFormValid()) {
      dispatch(startLoginWithEmailPassword(email, password));
    }
  };

  const handleGoogleLogin = () => {
    if (!loading) {
      dispatch(startGoogleLogin());
    }
  };

  const isFormValid = () => {
    if (!email || !password) {
      setError('All fields are required');
      return false;
    }

    if (!validator.isEmail(email)) {
      setError('Invalid email');
      return false;
    }

    if (password.length < 5) {
      setError('Password must be at least 5 characters');
      return false;
    }

    setError(undefined);
    return true;
  };

  return (
    <>
      <h3 className="auth__title mb-5">Login</h3>
      <form onSubmit={handleLogin}>
        {(error || msgError) && (
          <p className="alert-error mb-5">{error || msgError}</p>
        )}
        <input
          type="text"
          placeholder="Email"
          name="email"
          className="auth__input"
          value={email}
          onChange={handleInputChange}
        />
        <input
          type="password"
          placeholder="password"
          name="password"
          className="auth__input"
          autoComplete="off"
          value={password}
          onChange={handleInputChange}
        />
        <button
          disabled={loading}
          type="submit"
          className="btn btn-primary btn-block mb-5 mt-1"
        >
          Login
        </button>
        <div>
          <p className="mb-5">Login with social networks</p>

          <div className="google-btn mb-5" onClick={handleGoogleLogin}>
            <div className="google-icon-wrapper">
              <img
                className="google-icon"
                src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
                alt="google button"
              />
            </div>
            <p className="btn-text">
              <b>Sign in with google</b>
            </p>
          </div>
        </div>
        <Link className="link" to="/auth/register">
          Create new account
        </Link>
      </form>
    </>
  );
};
