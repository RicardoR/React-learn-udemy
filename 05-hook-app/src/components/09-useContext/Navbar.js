import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import './styles.css';

export const Navbar = () => {
  return (
    <nav className="navbar navbar-expand-md navbar-dark bg-dark">
      <Link className="navbar-brand" to="#">
        useContext
      </Link>

      <button
        class="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span class="navbar-toggler-icon"></span>
      </button>
      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="navbar-nav">
          <li className="nav-item">
            <NavLink className="nav-link" to="/">
              Home
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink className="nav-link" to="/about">
              About
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink className="nav-link" to="/login">
              Login
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
};
