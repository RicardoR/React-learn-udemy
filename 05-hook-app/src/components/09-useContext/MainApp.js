import React from 'react';
import { Outlet } from 'react-router-dom';
import { Navbar } from './Navbar';
import { UserContext } from './UserContext';

export const MainApp = () => {
  const user = {
    id: 123,
    name: 'Ricardo',
    email: 'email@email.com',
  };
  return (
    <div>
      <UserContext.Provider value={user}>
        <Navbar />
        {/* An <Outlet> renders whatever child route is currently active,
          so you can think about this <Outlet> as a placeholder for
          the child routes we defined above. */}
        <Outlet />
      </UserContext.Provider>
    </div>
  );
};
