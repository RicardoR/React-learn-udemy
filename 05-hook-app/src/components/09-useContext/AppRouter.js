import React from 'react';
import { Routes, Route } from 'react-router-dom';
import { AboutScreen } from './AboutScreen';
import { MainApp } from './MainApp';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
  return (
    <div>
      <Routes>
        <Route path="/" element={<MainApp />}>
          <Route index element={<HomeScreen />} />
          <Route path="about" element={<AboutScreen />} />
          <Route path="login" element={<LoginScreen />} />

          {/* Using path="*"" means "match anything", so this route
                acts like a catch-all for URLs that we don't have explicit
                routes for. */}
          <Route path="*" element={<HomeScreen />} />
        </Route>
      </Routes>
    </div>
  );
};
