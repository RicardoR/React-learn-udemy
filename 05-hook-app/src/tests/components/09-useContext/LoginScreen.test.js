import React from 'react';
import { mount } from 'enzyme';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('LoginScreen UT', () => {
  const user = {
    name: 'Jane Doe',
    email: 'jan@mail.com',
  };
  const setUser = jest.fn();

  // mount allow us to render the childs components
  const wrapper = mount(
    <UserContext.Provider value={{ user, setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test('should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should display user name', () => {
    expect(wrapper.find('h1').text()).toBe('LoginScreen');
    expect(wrapper.find('pre').text()).toBe(JSON.stringify(user, null, 3));
  });

  test('should handle login when button is clicked', () => {
    wrapper.find('button').simulate('click');
    expect(setUser).toHaveBeenCalledWith({
      id: 123,
      name: 'John Doe',
    });
  });
});
