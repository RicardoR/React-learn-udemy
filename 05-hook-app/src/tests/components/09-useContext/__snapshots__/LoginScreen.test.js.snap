// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen UT should display correctly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Jane Doe",
   "email": "jan@mail.com"
}
  </pre>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
